cookies test fails on Firefox 151 scaffolded to https://example.cypress.io

## Current

The cookies test spec [cookies.cy.js](https://github.com/cypress-io/cypress-example-kitchensink/blob/master/cypress/e2e/2-advanced-examples/cookies.cy.js) fails in Firefox 151.0 in the test

'cy.getAllCookies() - get all browser cookies'

when scaffolded in Cypress and running against https://example.cypress.io instead of http://localhost

## Steps to reproduce

Clone https://github.com/cypress-io/cypress-example-kitchensink

Modify 

https://github.com/cypress-io/cypress-example-kitchensink/blob/f3fd5e9956eafc56dabbca3f5b4b51026adb7304/cypress/e2e/2-advanced-examples/cookies.cy.js#L7

to test against https://example.cypress.io

Run using Firefox 151.0

## Expected

The test should run without error.

## Logs

**Failure with Firefox 151.0**

```text                                                                                              $ npx cypress run --browser firefox

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        15.15.0                                                                        │
  │ Browser:        Firefox 151 (headless)                                                         │
  │ Node Version:   v24.15.0 (/home/mike/n/bin/node)                                               │
  │ Specs:          1 found (cookies.cy.js)                                                        │
  │ Searched:       cypress/e2e/**/*.cy.{js,jsx,ts,tsx}                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  cookies.cy.js                                                                   (1 of 1)


  Cookies
  cypress:server:automation:cookies getting:cookies {} +0ms
  cypress:server:automation:cookies received get:cookies [] +3ms
  cypress:server:automation:cookies clear:cookies 0 +10ms
  cypress:server:automation:cookies getting:cookies { url: 'https://example.cypress.io/commands/cookies' } +14ms
  cypress:server:automation:cookies received get:cookies [] +4ms
  cypress:server:automation:cookies getting:cookies { domain: 'example.cypress.io' } +689ms
  cypress:server:automation:cookies received get:cookies [] +6ms
  cypress:server:automation:cookies getting:cookies {} +1s
  cypress:server:automation:cookies received get:cookies [] +42ms
  cypress:server:automation:cookies clear:cookies 0 +13ms
  cypress:server:automation:cookies getting:cookies { url: 'https://example.cypress.io/commands/cookies' } +14ms
  cypress:server:automation:cookies received get:cookies [] +41ms
  cypress:server:automation:cookies getting:cookies { domain: 'example.cypress.io' } +85ms
  cypress:server:automation:cookies received get:cookies [] +5ms
  cypress:server:automation:cookies getting:cookies { domain: 'example.cypress.io' } +809ms
  cypress:server:automation:cookies received get:cookies [] +2ms
  cypress:server:automation:cookies getting:cookies {} +21ms
  cypress:server:automation:cookies received get:cookies [] +1ms
  cypress:server:automation:cookies set:cookie { domain: 'example.cypress.io', httpOnly: false, hostOnly: false, name: 'key', path: '/', secure: false, value: 'value', expirationDate: 2410502063, url: 'http://example.cypress.io/' } +14ms
  cypress:server:automation:cookies received set:cookie { domain: '.example.cypress.io', httpOnly: false, name: 'key', path: '/', sameSite: 'unspecified', secure: false, value: 'value', expiry: 1813910064 } +18ms
  cypress:server:automation:cookies set:cookie { domain: '.example.com', httpOnly: false, hostOnly: false, name: 'key', path: '/', secure: false, value: 'value', expirationDate: 2410502064, url: 'http://example.com/' } +9ms
  cypress:server:automation:cookies received set:cookie { domain: '.example.com', httpOnly: false, name: 'key', path: '/', sameSite: 'unspecified', secure: false, value: 'value', expiry: 1813910064 } +8ms
  cypress:server:automation:cookies getting:cookies {} +9ms
  cypress:server:automation:cookies received get:cookies [ { domain: '.example.com', httpOnly: false, name: 'key', path: '/', sameSite: 'unspecified', secure: false, value: 'value', expiry: 1813910064 }, { domain: '.example.cypress.io', httpOnly: false, name: 'key', path: '/', sameSite: 'unspecified', secure: false, value: 'value', expiry: 1813910064 } ] +2ms
    1) cy.getAllCookies() - get all browser cookies


  0 passing (3s)
  1 failing

  1) Cookies
       cy.getAllCookies() - get all browser cookies:

      AssertionError: expected { Object (name, value, ...) } to have property 'domain' of '.example.com', but got '.example.cypress.io'
      + expected - actual

      -'.example.cypress.io'
      +'.example.com'

      at  (webpack://cypress-test-tiny/./cypress/e2e/cookies.cy.js:61:33)
      at __webpack_modules__</__WEBPACK_DEFAULT_EXPORT__/shouldFnWithCallback (https://example.cypress.io/__cypress/runner/cypress_runner.js:122366:17)
      at tryCatcher (https://example.cypress.io/__cypress/runner/cypress_runner.js:1777:23)
      at __webpack_modules__</module.exports/Promise.try (https://example.cypress.io/__cypress/runner/cypress_runner.js:4285:29)
      at shouldFnWithCallback (https://example.cypress.io/__cypress/runner/cypress_runner.js:122363:60)
      at shouldFn (https://example.cypress.io/__cypress/runner/cypress_runner.js:122406:37)
      at should (https://example.cypress.io/__cypress/runner/cypress_runner.js:122525:23)
      at wrapped (https://example.cypress.io/__cypress/runner/cypress_runner.js:146488:19)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     3 seconds                                                                        │
  │ Spec Ran:     cookies.cy.js                                                                    │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  /home/mike/github/CYPRESS-IO/cypress-test-tiny/cypress/screenshots/cookies.cy.js     (1280x720)
     /Cookies -- cy.getAllCookies() - get all browser cookies (failed).png


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖  cookies.cy.js                            00:03        1        -        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  1 of 1 failed (100%)                     00:03        1        -        1        -        -
```




## Related

- https://github.com/cypress-io/cypress/issues/33843

--- a/app/commands/cookies.html
+++ b/app/commands/cookies.html
@@ -132,22 +132,27 @@ cy.getCookies().should('have.length', 1).should((cookies) => {
 cy.setCookie('key', 'value')
 cy.setCookie('key', 'value', { domain: 'example.com' })
 
-// cy.getAllCookies() yields an array of cookies
+// cy.getAllCookies() yields an array of cookies (order is not guaranteed)
 cy.getAllCookies().should('have.length', 2).should((cookies) => {
-  // each cookie has these properties
-  expect(cookies[0]).to.have.property('name', 'key')
-  expect(cookies[0]).to.have.property('value', 'value')
-  expect(cookies[0]).to.have.property('httpOnly', false)
-  expect(cookies[0]).to.have.property('secure', false)
-  expect(cookies[0]).to.have.property('domain')
-  expect(cookies[0]).to.have.property('path')
+  const hostCookie = cookies.find((cookie) => cookie.domain !== '.example.com')
+  const exampleCookie = cookies.find((cookie) => cookie.domain === '.example.com')
 
-  expect(cookies[1]).to.have.property('name', 'key')
-  expect(cookies[1]).to.have.property('value', 'value')
-  expect(cookies[1]).to.have.property('httpOnly', false)
-  expect(cookies[1]).to.have.property('secure', false)
-  expect(cookies[1]).to.have.property('domain', 'example.com')
-  expect(cookies[1]).to.have.property('path')
+  // each cookie has these properties
+  expect(hostCookie).to.exist
+  expect(hostCookie).to.have.property('name', 'key')
+  expect(hostCookie).to.have.property('value', 'value')
+  expect(hostCookie).to.have.property('httpOnly', false)
+  expect(hostCookie).to.have.property('secure', false)
+  expect(hostCookie).to.have.property('domain')
+  expect(hostCookie).to.have.property('path')
+
+  expect(exampleCookie).to.exist
+  expect(exampleCookie).to.have.property('name', 'key')
+  expect(exampleCookie).to.have.property('value', 'value')
+  expect(exampleCookie).to.have.property('httpOnly', false)
+  expect(exampleCookie).to.have.property('secure', false)
+  expect(exampleCookie).to.have.property('domain', '.example.com')
+  expect(exampleCookie).to.have.property('path')
 })</code></pre>
         </div>
         <div class="col-xs-5">

--- a/cypress/e2e/2-advanced-examples/cookies.cy.js
+++ b/cypress/e2e/2-advanced-examples/cookies.cy.js
@@ -44,22 +44,27 @@ context('Cookies', () => {
     cy.setCookie('key', 'value')
     cy.setCookie('key', 'value', { domain: '.example.com' })
 
-    // cy.getAllCookies() yields an array of cookies
+    // cy.getAllCookies() yields an array of cookies (order is not guaranteed)
     cy.getAllCookies().should('have.length', 2).should((cookies) => {
-      // each cookie has these properties
-      expect(cookies[0]).to.have.property('name', 'key')
-      expect(cookies[0]).to.have.property('value', 'value')
-      expect(cookies[0]).to.have.property('httpOnly', false)
-      expect(cookies[0]).to.have.property('secure', false)
-      expect(cookies[0]).to.have.property('domain')
-      expect(cookies[0]).to.have.property('path')
+      const hostCookie = cookies.find((cookie) => cookie.domain !== '.example.com')
+      const exampleCookie = cookies.find((cookie) => cookie.domain === '.example.com')
 
-      expect(cookies[1]).to.have.property('name', 'key')
-      expect(cookies[1]).to.have.property('value', 'value')
-      expect(cookies[1]).to.have.property('httpOnly', false)
-      expect(cookies[1]).to.have.property('secure', false)
-      expect(cookies[1]).to.have.property('domain', '.example.com')
-      expect(cookies[1]).to.have.property('path')
+      // each cookie has these properties
+      expect(hostCookie).to.exist
+      expect(hostCookie).to.have.property('name', 'key')
+      expect(hostCookie).to.have.property('value', 'value')
+      expect(hostCookie).to.have.property('httpOnly', false)
+      expect(hostCookie).to.have.property('secure', false)
+      expect(hostCookie).to.have.property('domain')
+      expect(hostCookie).to.have.property('path')
+
+      expect(exampleCookie).to.exist
+      expect(exampleCookie).to.have.property('name', 'key')
+      expect(exampleCookie).to.have.property('value', 'value')
+      expect(exampleCookie).to.have.property('httpOnly', false)
+      expect(exampleCookie).to.have.property('secure', false)
+      expect(exampleCookie).to.have.property('domain', '.example.com')
+      expect(exampleCookie).to.have.property('path')
     })
   })
 

